reports/ccip-functions-report.md

```diff
@@ -2690,3 +2691,70 @@ Invoking `SSTORE`operations in loops may lead to Out-of-gas errors. Use a local
 
 
 
+## L-17: Potentially missing inheritance for contract.
+
+There is an interface / abstract contract that is potentially missing (not included in) the inheritance of this contract. If that's not the case, consider using the same interface instead of defining multiple identical interfaces.
+
+<details><summary>8 Found Instances</summary>
+
+
+- Found in src/v0.8/functions/dev/v1_X/FunctionsCoordinator.sol [Line: 13](../tests/ccip-contracts/contracts/src/v0.8/functions/dev/v1_X/FunctionsCoordinator.sol#L13)
+
+	Consider implementing the interface - IFunctionsCoordinator
+	```solidity
+	contract FunctionsCoordinator is OCR2Base, IFunctionsCoordinator, FunctionsBilling {
+	```
+
+- Found in src/v0.8/functions/dev/v1_X/FunctionsRouter.sol [Line: 16](../tests/ccip-contracts/contracts/src/v0.8/functions/dev/v1_X/FunctionsRouter.sol#L16)
+
+	Consider implementing the interface - IFunctionsRouter
+	```solidity
+	contract FunctionsRouter is IFunctionsRouter, FunctionsSubscriptions, Pausable, ITypeAndVersion, ConfirmedOwner {
+	```
+
+- Found in src/v0.8/functions/dev/v1_X/accessControl/TermsOfServiceAllowList.sol [Line: 14](../tests/ccip-contracts/contracts/src/v0.8/functions/dev/v1_X/accessControl/TermsOfServiceAllowList.sol#L14)
+
+	Consider implementing the interface - ITermsOfServiceAllowList
+	```solidity
+	contract TermsOfServiceAllowList is ITermsOfServiceAllowList, IAccessController, ITypeAndVersion, ConfirmedOwner {
+	```
+
+- Found in src/v0.8/functions/v1_0_0/FunctionsCoordinator.sol [Line: 13](../tests/ccip-contracts/contracts/src/v0.8/functions/v1_0_0/FunctionsCoordinator.sol#L13)
+
+	Consider implementing the interface - IFunctionsCoordinator
+	```solidity
+	contract FunctionsCoordinator is OCR2Base, IFunctionsCoordinator, FunctionsBilling {
+	```
+
+- Found in src/v0.8/functions/v1_0_0/FunctionsRouter.sol [Line: 16](../tests/ccip-contracts/contracts/src/v0.8/functions/v1_0_0/FunctionsRouter.sol#L16)
+
+	Consider implementing the interface - IFunctionsRouter
+	```solidity
+	contract FunctionsRouter is IFunctionsRouter, FunctionsSubscriptions, Pausable, ITypeAndVersion, ConfirmedOwner {
+	```
+
+- Found in src/v0.8/functions/v1_1_0/FunctionsCoordinator.sol [Line: 13](../tests/ccip-contracts/contracts/src/v0.8/functions/v1_1_0/FunctionsCoordinator.sol#L13)
+
+	Consider implementing the interface - IFunctionsCoordinator
+	```solidity
+	contract FunctionsCoordinator is OCR2Base, IFunctionsCoordinator, FunctionsBilling {
+	```
+
+- Found in src/v0.8/functions/v1_3_0/FunctionsCoordinator.sol [Line: 13](../tests/ccip-contracts/contracts/src/v0.8/functions/v1_3_0/FunctionsCoordinator.sol#L13)
+
+	Consider implementing the interface - IFunctionsCoordinator
+	```solidity
+	contract FunctionsCoordinator is OCR2Base, IFunctionsCoordinator, FunctionsBilling {
+	```
```

Comment:
ooo Interesting case ...

@alexroan Help me here please. What shall we do in this case?

Here, there are 2 interfaces named IFunctionsCoordinator . That's why this has been caught 
<img width="761" alt="Screenshot 2024-08-31 at 12 26 41 PM" src="https://github.com/user-attachments/assets/aaaa0ea8-cab1-4666-8238-72f06dc24da6">

Reply:
Or should we assume the re-used contract name detector catches that and therefore the dev has resolved it ?

Reply:
UPDATE: I'm going to restrict this detector to run on only those contracts that don't inherit from anything